app-android/src/main/java/com/ustadmobile/port/android/view/SchoolDetailOverviewFragment.kt

```diff
@@ -170,4 +194,174 @@ class SchoolDetailOverviewFragment: UstadDetailFragment<SchoolWithHolidayCalenda
+                labelText = stringResource(id = R.string.school_code),
+                imageId = R.drawable.ic_login_24px,
+                onClick = onClickSchoolCode
+            )
+        }
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        if (uiState.schoolAddressVisible){
+            UstadDetailField(
+                valueText = uiState.entity?.schoolAddress ?: "",
+                labelText = stringResource(id = R.string.address),
+                imageId = R.drawable.ic_location_pin_24dp
+            )
+        }
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        if (uiState.schoolPhoneNumberVisible){
+            UstadDetailField(
+                valueText = uiState.entity?.schoolPhoneNumber ?: "",
+                labelText = stringResource(id = R.string.phone_number),
+                imageId = R.drawable.ic_call_bcd4_24dp,
+                onClick = onClickSchoolPhoneNumber,
+                secondaryActionContent = {
+                    IconButton(
+                        onClick = onClickSms,
+                    ) {
+                        Icon(
+                            imageVector = Icons.Filled.Message,
+                            contentDescription = stringResource(id = R.string.message),
+                        )
+                    }
+                }
+            )
+        }
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        if (uiState.calendarUidVisible){
+            UstadDetailField(
+                valueText = uiState.entity?.holidayCalendar?.umCalendarName ?: "",
+                labelText = stringResource(id = R.string.holiday_calendar),
+                imageId = R.drawable.ic_perm_contact_calendar_black_24dp
+            )
+        }
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        if (uiState.schoolEmailAddressVisible){
+            UstadDetailField(
+                valueText = uiState.entity?.schoolAddress ?: "",
+                labelText = stringResource(id = R.string.email),
+                imageId = R.drawable.ic_email_black_24dp,
+                onClick = onClickEmail
+            )
+        }
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        if (uiState.schoolTimeZoneVisible){
+            UstadDetailField(
+                valueText = uiState.entity?.schoolTimeZone ?: "",
+                labelText = stringResource(id = R.string.timezone),
+                imageId = R.drawable.ic_language_blue_grey_600_24dp,
+            )
+        }
+
+        Spacer(modifier = Modifier.height(15.dp))
+
+        Text(text = stringResource(id = R.string.classes),
+            style = Typography.h6
+        )
+
+        Clazzes(
+            clazzes = uiState.clazzes,
+            onClickClazz = onClickClazz
+        )
+    }
+}
+
+
+@Composable
+private fun Clazzes(
+    clazzes: Map<Int, ClazzWithListDisplayDetails>,
+    onClickClazz: () -> Unit
+){
+
+    clazzes.forEach { (key, value) ->
+        TextButton(onClick = onClickClazz) {
```

Comment:
We shouldn't use a button as the base component here. It will change the styles (e.g. alignment, text color, etc). Please use the normal column and row, and then use the modifier.clickable for the onClick function. This will also provide the ripple effect.

Reply:
Also we should use the Jetpack compose official ListItem to manage the styles (even though it is experimental): https://developer.android.com/reference/kotlin/androidx/compose/material/package-summary#ListItem(androidx.compose.ui.Modifier,kotlin.Function0,kotlin.Function0,kotlin.Boolean,kotlin.Function0,kotlin.Function0,kotlin.Function0)

Reply:
e.g. 

```
ListItem(
   modifier = Modifier.clickable {
      onClickClazz(clazz)
    },
    text = { Text(clazz.clazzName ?: "") },
    secondaryText = { Text(clazz.clazzDesc ?: "") }  
)
```